### PR TITLE
RCS 2.0 - file roles store support for remote indices

### PR DESCRIPTION
**Pull request by n1v0lg**
Adds support for specifying `remote_indices` privileges in the file roles store.

This is just a matter of removing the safe-guard that previously prevented this, and adding tests.

Note: there is no need for a feature flag check; that's already handled by `RoleDescriptor.parse` which is used internally by `FileRolesStore`.

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elasticsearchmachine run elasticsearch-ci/release-tests

**Comment by n1v0lg** [comment]
Failure is unrelated. I opened an issue: https://github.com/elastic/elasticsearch/issues/91505

File: x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authz/store/roles_with_remote_indices.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        grant:
+          - foo
+          - boo
+        except:
+          - boo
+
+invalid_role_missing_clusters:
+  remote_indices:
+    - names:
+        - idx1
```
Comment (justincr-elastic) [inline x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authz/store/roles_with_remote_indices.yml:31]: Is it possible to test for empty names?

File: x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authz/store/roles_with_remote_indices.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+role_with_dls_fls:
+  remote_indices:
+    - clusters:
+        - "*"
+      names:
+        - idx1
+      privileges:
+        - READ
+      query: '{ "match_all": {} }'
```
[the remark this answers is not in the document]
Comment (n1v0lg) [reply x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authz/store/roles_with_remote_indices.yml:20]: Yup, good catch!

**Comment by n1v0lg** [comment]
@elasticsearchmachine run elasticsearch-ci/release-tests

**Review by justincr-elastic — APPROVED** [review]

**Pull request merged**